### PR TITLE
[Narwhal] compute and log ConsensusOutput digest

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Currently there is no simple way to check the consistency of ConsensusOutput among validator. The integrity hash is only calculated after some processings in consensus handler. Add a way to calculate ConsensusOutput digest and log it for every consensus commit.

After having more confidence in the value, we can commit to the digest in CheckpointSummary.

## Test Plan 

CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by arun-koshy — APPROVED** [review]

File: crates/sui-core/src/consensus_handler.rs [hunk 1]
```diff
@@ -191,14 +194,21 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
             .consensus_committed_subdags
             .with_label_values(&[&leader_author.to_string()])
             .inc();
-        for (cert, batches) in consensus_output.batches {
+        for (cert, batches) in consensus_output
+            .sub_dag
+            .certificates
+            .iter()
+            .zip(consensus_output.batches.iter())
```
Comment (akichidis) [inline crates/sui-core/src/consensus_handler.rs:201]: Is our intention to eventually remove the `certificate` from the `batches` vector? Otherwise why this change?

File: narwhal/types/src/consensus.rs [hunk 2]
```diff
@@ -21,7 +21,21 @@ pub type SequenceNumber = u64;
 /// It is sent to the the ExecutionState handle_consensus_transactions
 pub struct ConsensusOutput {
     pub sub_dag: Arc<CommittedSubDag>,
-    pub batches: Vec<(Certificate, Vec<Batch>)>,
+    /// Matches certificates in the `sub_dag` one-to-one.
+    pub batches: Vec<Vec<Batch>>,
```
Comment (akichidis) [inline narwhal/types/src/consensus.rs:25]: Nice 👍

**Review by akichidis — COMMENTED** [review]
Very nice! 

> After having more confidence in the value, we can commit to the digest in CheckpointSummary.


I would also add some testing on that on upcoming PR especially if we consider adding this to the CheckpointSummary, as an "innocent" update to the ConsensusOutput could break things.
